lib/Adapters/RegressionAdapter/hlsChecksumRegressionValidator.js

```diff
@@ -0,0 +1,300 @@
+   period.
+ */
+class HlsChecksumRegressionValidator extends RegressionValidatorBase {
+
+    constructor(config, flavors, last_chunklist_index) {
+        super(config, flavors, last_chunklist_index);
+
+        this.logger = logger.getLogger('hlsChecksumRegressionValidator');
+        this.override = config.validator.override;
+        this.last_chunklist_index = last_chunklist_index;
+        // path to load/save regression test results
+        // the path is initialized in init()
+        this.fullpath = config.validator.path + '/' + config.entryPath;
+        if (config.entryPath.lastIndexOf('/') < this.fullpath.length - 1) {
+            this.fullpath += '/';
+        }
+        // add root to path
+        this.db_filename = this.fullpath + this.entryId + '_regression_test_db.json';
+
+        // create initial json for regression results database
+        this.regression_results_db;
+        this.this_run_results = {};
+    }
+
+    init() {
+        var that = this;
+        return this.loadResultsDatabase()
+            .then(function () {
+                that.logger.debug('[%s] regression tests validator successfully initialized.', that.entryId);
+            })
+            .catch(function (err) {
+                that.logger.error('[%s] failed to initialize regression tests validator. Error: %s', that.entryId, ErrorUtils.error2string(err));
+            });
+    }
+
+    validateAllFlavors(index) {
+        let p = _.map(this.flavors, (next_flavor) => {
+            return this.validateSingleFlavor(this.entryId, next_flavor, index);
+        });
+
+        return Q.all(p)
+    }
+
+}
+
+
+HlsChecksumRegressionValidator.prototype.createDestinationDir = function() {
+
+    var that = this;
+ 
+    return qio.makeTree(that.fullpath)
+        .then( function() {
+            that.logger.debug('[%s] successfully created dir to save regression test results, [dir=%s].', that.entryId, that.fullpath);
+        })
+        .catch (function(err) {
+            that.logger.error('[%s] error creating dir to save regression test results, [dir=%s]. Error: %s', that.entryId, that.fullpath, ErrorUtils.error2string(err));
+        });
+}
+
+HlsChecksumRegressionValidator.prototype.getSegmentFromPackager = function(item, data) {
+
+    var that = this;
+    var deferred  = Q.defer();
+
+    data.errors[data.flavor] = {errors: {}};
+    var flavor_header = logheader1 `${this.entryId}${data.flavor}${data.index}`;
+
+    //let url = util.format("http://localhost:8080/kLive/smil:%s_all.smil/%s/%s", this.entryId, data.flavor, item.properties.uri);
+    let options = {
+        method: 'HEAD',
+        url: `http://localhost:8080/kLive/smil:${this.entryId}_all.smil/${data.flavor}/${item.properties.uri}`,
+       
+        timeout: 10000};
+    networkClient.read(options)
+        .then(function (content) {
+            // add to results new segment
+            var ts_id = item.properties.uri.split('.')[0];
```

Comment:
done. The length is taken from content-length header filed. It required also update of NetworkClient to retune respose.headers instead of body, when method: HEAD is part of request.